### ISSUE TITLE
Tests fail with Python 3.12

### ISSUE THREAD

**Issue by gvanrossum** [python/pyperformance#263]
Whenever I push something to a PR in the cpython main branch (3.12) I get an email from this repo telling me that the 3.12 run failed. Is there a way we can fix it? Here's a link to the failing run: https://github.com/python/pyperformance/actions/runs/4094033809/jobs/7059879010

I'd rather not just disable running with 3.12 (that would presumably take away some useful signal) but maybe we can fix or suppress the failing tests?

**Comment by AlexWaygood** [comment]
I believe this PR by @hugovk would mean that the tests would still run, but wouldn't be reported as "failures" for now:

- #259

**Comment by gvanrossum** [comment]
Yeah, that just disables 3.12 altogether, right? Isn't there a way to only disable those tests that depend on greenlet?

**Comment by gvanrossum** [comment]
IIRC in the past we resolved a similar thing (for a different dependency) by changing the requirements.txt file to reference a specific PR (via some notation I've forgotten that lets you depend on a git repo/branch).

**Comment by hugovk** [comment]
> Yeah, that just disables 3.12 altogether, right?

Yes, the original intent of #259 was to run on 3.12, and if 3.12 fails, then don't fail the whole build. But I couldn't find the right `continue-on-error ` incantation, so instead it disables the whole 3.12 job.

> Isn't there a way to only disable those tests that depend on greenlet?

Yes, we can skip them when greelet isn't available. Please see PR https://github.com/python/pyperformance/pull/264.

**Comment by gvanrossum** [comment]
Yeah, #264 feels better. I hesitate to approve it, since I am not familiar with this code base (nor with the sutbleties of GitHub workflows) but maybe @mdboom can take a quick look.

**Comment by corona10** [comment]
@gvanrossum I proposed another PR that does not touch pyperformance code directly.

PR: https://github.com/python/pyperformance/pull/274 (Expect as working correctly after the PR is merged)
Sample: https://github.com/corona10/pyperformance/pull/1#issuecomment-1514508151

**Comment by corona10** [comment]
Ah https://github.com/python/pyperformance/pull/259 looks similar I missed it

**Comment by hugovk** [comment]
Looks like the key difference between those, you have `continue-on-error: ${{ matrix.experimental }}` at the job level, I had it at the step level. Nice work!

**Comment by corona10** [comment]
Close this issue :) If the issue is needed to be reopened, feel free to open the issue again.
